presto-main/src/main/java/com/facebook/presto/operator/aggregation/AverageAggregations.java

```diff
@@ -32,14 +32,28 @@
     private AverageAggregations() {}
 
     @InputFunction
-    public static void input(LongAndDoubleState state, @SqlType(StandardTypes.BIGINT) long value)
+    public static void bigintInput(LongAndDoubleState state, @SqlType(StandardTypes.BIGINT) long value)
     {
         state.setLong(state.getLong() + 1);
         state.setDouble(state.getDouble() + value);
     }
 
     @InputFunction
-    public static void input(LongAndDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
+    public static void doubleInput(LongAndDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
+    {
+        state.setLong(state.getLong() + 1);
```

Comment:
We would actually want to make our own version that threw `PrestoException` with an error code. We probably want to benchmark to make sure that doesn't hurt performance though, and then change it in all the aggregations. Can you file a ticket for that?